components/expression/private/condition.jsx

```diff
@@ -0,0 +1,239 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Expression Condition design pattern](https://lightningdesignsystem.com/components/visual-picker/) in React.
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+import { EXPRESSION_CONDITION } from '../../../utilities/constants';
+
+import Combobox from '../../combobox';
+import Input from '../../input';
+import Button from '../../button';
+
+const propTypes = {
+	assistiveText: PropTypes.shape({
+		title: PropTypes.string,
+		delete: PropTypes.string,
+	}),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * CSS classes to be added to tag with `.slds-form-element`. Uses `classNames` [API](https://github.com/JedWatson/classnames).
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+
+	labels: PropTypes.shape({
+		condition: PropTypes.string,
+		resource: PropTypes.string,
+		operator: PropTypes.string,
+		value: PropTypes.string,
+	}),
+
+	resourcesList: PropTypes.arrayOf(PropTypes.object),
+	resourceSelected: PropTypes.object,
+	onChangeResource: PropTypes.func,
+
+	operator: PropTypes.oneOf(['equal', 'notEqual', 'greaterThan', 'lessThan']),
+	onChangeOperator: PropTypes.func,
+
+	onDelete: PropTypes.func,
+
+	value: PropTypes.string,
+	onChangeValue: PropTypes.func,
+
+	isChild: PropTypes.bool,
+};
+
+const operatorSelections = [
```

Comment:
data can't be in a LDSR component or private component like this as the developer can't modify this code for their apps